### PR TITLE
Update README with TCP deprecation notes

### PR DESCRIPTION
**Pull request by rsowald**

### CODE REVIEW

**Review by cdevarenne — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -28,13 +29,13 @@ Then create the certificates for the tests in the `certs` directory:
...
 
-We have a community discussion space at [Event Store Discuss](https://discuss.eventstore.com/). If you prefer slack, there is also an #eventstore channel in the [DDD-CQRS-ES](https://j.mp/ddd-es-cqrs) slack community.
+Interact with the Event Store and event sourcing Communities on the Event Store [Discuss](https://discuss.eventstore.com/) or [Discord](https://discord.gg/Phn9pmCw3t) forum.
 
 ## Contributing
 
 Development is done on the `master` branch.  
-We attempt to do our best to ensure that the history remains clean and to do so, we generally ask contributors to squash their commits into a set or single logical commit.
\ No newline at end of file
+We attempt to do our best to ensure that the history remains clean and to do so, we generally ask contributors to squash their commits into a set or single logical commit.
```
Comment (rmalik-es) [inline README.md:41]:
```suggestion
To maintain a clean history, we ask contributors to squash their commits into one or a few cohesive commits.
```

File: README.md [hunk 2]
```diff
@@ -2,8 +2,9 @@
 
 EventStoreDB is the open-source, functional database with Complex Event Processing in JavaScript.
 
-This is the repository for the legacy .NET TCP client for EventStoreDB version 5+ and uses TCP as the communication protocol.  
-If you're looking for the new recommended .NET gRPC client, check the [EventStore-Client-Dotnet](https://github.com/EventStore/EventStore-Client-Dotnet) repo.
+This is the repository for the legacy .NET TCP client for EventStoreDB version 5 - 23.10 and uses TCP as the communication protocol. ESDB versions later than 23.10 do not support TCP.
+
+If you're looking for the recommended .NET gRPC client, check the [EventStore-Client-Dotnet](https://github.com/EventStore/EventStore-Client-Dotnet) repo.
```
Comment (rmalik-es) [inline README.md:7]:
```suggestion
For the preferred .NET gRPC client, visit the [EventStore-Client-Dotnet](https://github.com/EventStore/EventStore-Client-Dotnet) repository.
```

**Review by rmalik-es — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -28,13 +29,13 @@ Then create the certificates for the tests in the `certs` directory:
 
 ## Documentation
 
-Documentation for EventStoreDB can be found here: [Event Store Docs](https://eventstore.com/docs/).
+Documentation for EventStoreDB can be found here: [Event Store Docs](https://developers.eventstore.com/).
 
 ## Community
 
-We have a community discussion space at [Event Store Discuss](https://discuss.eventstore.com/). If you prefer slack, there is also an #eventstore channel in the [DDD-CQRS-ES](https://j.mp/ddd-es-cqrs) slack community.
+Interact with the Event Store and event sourcing Communities on the Event Store [Discuss](https://discuss.eventstore.com/) or [Discord](https://discord.gg/Phn9pmCw3t) forum.
```
Comment (rmalik-es) [inline README.md:36]:
```suggestion
Interact with the Event Store and event sourcing communities on the Event Store [Discuss](https://discuss.eventstore.com/) or [Discord](https://discord.gg/Phn9pmCw3t) forum.
```

**Review by Hiyatei — APPROVED** [review]
LGTM, thanks for updating links on the way through!

**Review by cdevarenne — APPROVED** [review]

**Review by rmalik-es — APPROVED** [review]

**Review by thefringeninja — APPROVED** [review]

**Pull request merged**